examples/java/order.groovy

```diff
@@ -0,0 +1,237 @@
+        long expiry = timestamp + 24 * 60 * 60 // now + 24 hours
+
+        // Chain ID - PRIVATE_SN_POTC_SEPOLIA
+        BigInteger chainID = new BigInteger('7693264728749915528729180568779831130134670232771119425')
+        String chainIdHex = '0x' + chainID.toString(16).toUpperCase()
+
+        // Create the auth message
+        def authMessage = createAuthMessage(timestamp, expiry, chainIdHex)
+
+        // Get signature
+        def (String signatureStr, String messageHashStr) = getSignature(authMessage)
+
+        // Call the auth endpoint
+        def requestHeaders = [
+            'PARADEX-STARKNET-ACCOUNT': accountAddressStr,
+            'PARADEX-STARKNET-SIGNATURE': signatureStr,
+            'PARADEX-STARKNET-MESSAGE-HASH': messageHashStr,
+            'PARADEX-TIMESTAMP': timestamp.toString(),
+            'PARADEX-SIGNATURE-EXPIRATION': expiry.toString(),
+        ]
+        def jsonResponse = postRestApi("${apiUrl}/auth", requestHeaders, null)
+        def jwtToken = jsonResponse.jwt_token
+        return jwtToken
+    }
+
+    static getSignature(String message) {
+        // Convert the account address and private key to Felt types
+        Felt accountAddress = Felt.fromHex(accountAddressStr)
+        Felt privateKey = Felt.fromHex(privateKeyStr)
+
+        // Convert the message to a typed data object
+        TypedData typedData = TypedData.fromJsonString(message)
+        Felt messageHash = typedData.getMessageHash(accountAddress)
+        String messageHashStr = messageHash.hexString().toString()
+
+        // Create new StarkCurveSigner with the private key
+        StarkCurveSigner scSigner = new StarkCurveSigner(privateKey)
+
+        // Sign the typed data
+        List<Felt> signature = scSigner.signTypedData(typedData, accountAddress)
+
+        // Convert the signature to a string
+        List<BigInteger> signatureBigInt = signature.collect { it.getValue().toBigInteger() }
+        def signatureStr = convertBigIntListToString(signatureBigInt)
+        return [signatureStr, messageHashStr]
+    }
+
+    static Object postRestApi(String url, Map headers, LinkedHashMap payload) {
+        println "Request: POST ${url}"
+        def http = new URL(url).openConnection() as HttpURLConnection
+        headers.each { key, value ->
+            http.setRequestProperty(key, value)
+        }
+        http.setRequestProperty("Content-Type", "application/json; charset=UTF-8");
+        http.setRequestMethod('POST')
+        http.setDoOutput(true)
+
+        if (payload != null) {
+            def postBody = new groovy.json.JsonBuilder(payload).toPrettyString()
+            println "Request Body: ${postBody}"
+            http.outputStream.withWriter { writer ->
+                writer.write(postBody)
+            }
+        }
+        http.connect()
+
+        if (http.getResponseCode() >= 400) {
+            println "Error: ${http.getErrorStream().getText("UTF-8")}"
```

Comment:
What is returned in this case? In null, then we could potentially have an issue on getJWT() since we are directly doing `jsonResponse.jwt_token` without any verification?